### ISSUE TITLE
event bus trigger wrong handler with multiple instances of a presenter

### ISSUE THREAD

**Issue by renfeng** [panterch/org.vaadin.mvp#2]
hello. i had the following code to create a list of items. each item has its presenter instances. when i clicked on an item, the event was only sent to the last instance of presenters. however, i expect the presenter instance of the item being clicked should receive the event. please have a look at this. see if this was a flaw. now i have a workaround by storing presenter instance into the button's data , 

button.setData(oneOfMyPresenterInstance);

here goes the code building my list, thanks:

```
        ComponentContainer contactContainer = view.getContactContainer();
        contactContainer.removeAllComponents();

        contactFlows = new ArrayList<ViewFlow>(contacts.size());
        for (Contact c : contacts) {

            ViewFlow contactFlow = new ViewFlow(contactContainer);

            ContactPresenter contactPresenter = (ContactPresenter) appRunner
                    .createPresenter(ContactPresenter.class);
            contactPresenter.setContact(c);
            // contactPresenter.setPresenter(this);
            contactPresenter.start(appRunner, contactFlow);

            contactFlows.add(contactFlow);
        }
```

**Comment by apalumbo** [comment]
Hello, the issue is related to the sharing of the bus between the instances of the presenter. I have provided a patch for that issue but to enable that behaviour on a bus you need to put an annotation on the bus interface @PrivateBus.
There is already a pull request.

**Comment by mtam** [comment]
Should be fixed with new version 0.9.2 with contributions from Alessandro.
